### PR TITLE
Update `amax/amin/norm/count_nonzero` signatures with `int[*]? dim`

### PR DESCRIPTION
**Pull request by kurtamohler**
Changes `dim` arg to use `int[*]?` type for the following functions in `native_funcitons.yaml`:
* `amax`
* `amin`
* `norm`
* `frobenius_norm`
* `native_norm`
* `count_nonzero`

Part of #29137

### CODE REVIEW

**Review by ngimel — APPROVED** [review]
Looks good, let's see what tests say.

**Comment by kurtamohler** [comment]
@wonjoolee95 / @JackCaoG, this PR breaks a couple more signatures for XLA

**Comment by JackCaoG** [comment]
@wonjoolee95 I will let you take care of this one.

File: test/forward_backward_compatibility/check_forward_backward_compatibility.py
Comment (albanD) [inline test/forward_backward_compatibility/check_forward_backward_compatibility.py:146]: note that if you want to reduce the merge conflicts on this file, you can add your changes in the middle of it ;)

**Review by albanD — APPROVED** [review]
This needs rebasing and xla hash update once the fix lands there!

File: test/forward_backward_compatibility/check_forward_backward_compatibility.py
Comment (kurtamohler) [reply test/forward_backward_compatibility/check_forward_backward_compatibility.py:146]: Oh good to know, I'll do that in the future

**Comment by wonjoolee95** [comment]
Hey @kurtamohler, can we rebase this PR on master? I'm having some issues building this PR and PT/XLA master locally due to differences in commits. Thanks!

**Comment by kurtamohler** [comment]
@wonjoolee95, rebase is done

**Review by kulinseth — APPROVED** [review]

**Comment by wonjoolee95** [comment]
Opened https://github.com/pytorch/xla/pull/4035 from XLA's side, now pending CIs.

**Comment by wonjoolee95** [comment]
@kurtamohler, sorry to ask again but can we rebase this with master one more time? The XLA PR CI is failing due to out-of-date changes.

**Comment by kurtamohler** [comment]
No worries @wonjoolee95! Rebase is done

**Comment by kurtamohler** [comment]
@wonjoolee95, just FYI, I had to make a small update to fix a newly failing CI check

**Comment by wonjoolee95** [comment]
@kurtamohler, the XLA changes looking good. Tests are succeeding locally. The new XLA hash should be `f3c6a8fb31e61e8170140de1a93b2cb48b1833c6`. When you update the hash pin, if you could rebase this PR once again (hopefully the last time 😄 ), that'd be great. Thanks!

**Comment by kurtamohler** [comment]
@wonjoolee95 thanks! I've added the hash and rebased

**Comment by wonjoolee95** [comment]
Thanks, @kurtamohler! Once the CIs are green on both sides, we should be good to merge these.

**Comment by huydhn** [comment]
@pytorchbot revert -m 'The commit pin breaks XLA test somehow' -c weird

```
2022-09-28T00:58:19.9365256Z + clone_pytorch_xla
2022-09-28T00:58:19.9365529Z + [[ ! -d ./xla ]]
2022-09-28T00:58:19.9365831Z + git clone --recursive --quiet https://github.com/pytorch/xla.git
2022-09-28T00:59:13.0765957Z + pushd xla
2022-09-28T00:59:13.0766488Z /var/lib/jenkins/workspace/xla /var/lib/jenkins/workspace
2022-09-28T00:59:13.0768146Z ++ cat ../.github/ci_commit_pins/xla.txt
2022-09-28T00:59:13.0783190Z + git checkout bb7830f88646eb502ecbc52e3824d48826fa212d
2022-09-28T00:59:13.0797587Z fatal: reference is not a tree: bb7830f88646eb502ecbc52e3824d48826fa212d
```

It looks like the change is reverted on XLA

**Comment by JackCaoG** [comment]
hey.. this is a mistake, we merge the change accidentlly too early. We revert it and revert it again in https://github.com/pytorch/xla/commit/ec3fd75b2d4bbcf62917d0818378aa3a7d0edcb7 . There is no need to revert this pr

**Comment by JackCaoG** [comment]
Can we merge this pr again? Otherwise I have to revert the revert of revert, which is pretty annoying

**Comment by huydhn** [comment]
> Can we merge this pr again? Otherwise I have to revert the revert of revert, which is pretty annoying

I see. Once you are ready, please just force merge this one then to put it back into trunk

**Comment by JackCaoG** [comment]
@huydhn I don't have the write access to this pr, can you update `xla.txt` to `ec3fd75b2d4bbcf62917d0818378aa3a7d0edcb7` and force a merge? this is the commit of new head

**Comment by huydhn** [comment]
Hmm, it looks like I couldn't update it neither.  Let me see if I miss anything here

File: .github/ci_commit_pins/xla.txt
Comment (huydhn) [inline .github/ci_commit_pins/xla.txt:1]:
```suggestion
ec3fd75b2d4bbcf62917d0818378aa3a7d0edcb7
```

**Comment by kurtamohler** [comment]
Thanks for the fix @huydhn !

**Comment by jeffdaily** [comment]
Trying to confirm if this is breaking ROCm periodic tests, specifically nvfuser.

```
RuntimeError: isIntList() INTERNAL ASSERT FAILED at "/var/lib/jenkins/workspace/aten/src/ATen/core/ivalue_inl.h":1833, please report a bug to PyTorch. Expected IntList but got None
```

Raw logs:
https://ossci-raw-job-status.s3.amazonaws.com/log/8585934815

**Comment by kurtamohler** [comment]
I will look into fixing this and submit a new PR

**Pull request closed without merging**
